plugins/mongo_db/mongo_db_operations.cpp

```diff
@@ -0,0 +1,819 @@
+#include <golos/plugins/mongo_db/mongo_db_operations.hpp>
+#include <bsoncxx/builder/stream/array.hpp>
+#include <bsoncxx/builder/stream/value_context.hpp>
+#include <bsoncxx/builder/basic/kvp.hpp>
+#include <bsoncxx/builder/basic/document.hpp>
+
+namespace golos {
+namespace plugins {
+namespace mongo_db {
+
+    using bsoncxx::builder::stream::array;
+    using bsoncxx::builder::basic::kvp;
+    using bsoncxx::builder::stream::document;
+
+    // Helper functions
+    document format_authority(const authority& auth) {
+        array account_auths_arr;
+        for (auto iter : auth.account_auths) {
+            document temp;
+            temp << "public_key_type" << (std::string)iter.first
+                 << "weight_type" << std::to_string(iter.second);
+            account_auths_arr << temp;
+        }
+        array key_auths_arr;
+        for (auto iter : auth.key_auths) {
+            document temp;
+            temp << "public_key_type" << (std::string)iter.first
+                 << "weight_type" << std::to_string(iter.second);
+            key_auths_arr << temp;
+        }
+
+        document authority_doc;
+        authority_doc << "owner" << auth.owner
+                      << "weight_threshold" << std::to_string(auth.weight_threshold);
+        authority_doc << "account_auths" << account_auths_arr;
+        authority_doc << "key_auths" << key_auths_arr;
+
+        return authority_doc;
+    }
+
+    document format_chain_properties(const chain_properties& props) {
+        document props_doc;
+        props_doc << "account_creation_fee" << props.account_creation_fee.to_string()
+                  << "maximum_block_size" << std::to_string(props.maximum_block_size)
+                  << "sbd_interest_rate" << std::to_string(props.sbd_interest_rate);
+        return props_doc;
+    }
+
+    std::string to_string(const signature_type& signature) {
+        std::string retVal;
+
+        for (auto iter : signature) {
+            retVal += iter;
+        }
+        return retVal;
+    }
+    /////////////////////////////////////////////////
+
+
+    document write_operation(const operation& op) {
+        return document{};
```

Comment:
Guys i have a question. How do i extract proper operation from static_variant? So far the only thought i got is to cal .where() - it is a position of type that is currently in operation, and then to deduce type from declaration of operation static variant., and then cal .as<Type>() and get what i want. This does not feels like right tho. Any ideas?